### PR TITLE
try to get display columns to go in and out on state change

### PR DESCRIPTION
**Pull request by naughton**
let me know if i'm going down the wrong path here...

### CODE REVIEW

**Comment by posabsolute** [comment]
Personally I like it, looks cleaner

**Comment by alessandrojcm** [comment]
Me too, way easier to follow

File: packages/mantine-react-table/src/hooks/useMRT_DisplayColumns.tsx [hunk 1]
```diff
@@ -21,135 +23,169 @@ interface Params<TData extends Record<string, any> = {}> {
...
-      tableOptions.localization,
-      tableOptions.positionActionsColumn,
-      tableOptions.renderDetailPanel,
-      tableOptions.renderRowActionMenuItems,
-      tableOptions.renderRowActions,
-      tableOptions.state?.columnOrder,
-      tableOptions.state?.grouping,
-    ],
-  );
+export const useMRT_DisplayColumns = <TData extends Record<string, any> = {}>(
```
Comment (KevinVandy) [inline packages/mantine-react-table/src/hooks/useMRT_DisplayColumns.tsx:26]: I really like the refactoring done here

File: packages/mantine-react-table/src/hooks/useMRT_TableInstance.ts [hunk 2]
```diff
@@ -55,8 +55,8 @@ export const useMRT_TableInstance: <TData extends Record<string, any> = {}>(
 
   const initialState: Partial<MRT_TableState<TData>> = useMemo(() => {
     const initState = tableOptions.initialState ?? {};
-    initState.columnOrder =
-      initState.columnOrder ?? getDefaultColumnOrderIds(tableOptions);
+    // initState.columnOrder =
+    //   initState.columnOrder ?? getDefaultColumnOrderIds(tableOptions);
```
Comment (KevinVandy) [inline packages/mantine-react-table/src/hooks/useMRT_TableInstance.ts]: but a bit concerned about what's going on here

Comment (KevinVandy) [reply packages/mantine-react-table/src/hooks/useMRT_TableInstance.ts]: The problem introduced here is that this is simply turning off the column ordering feature entirely within MRT. That's why dynamic columns seem to start working. This means though that when column ordering is turned on by the developer, everything breaks again.

Comment (naughton) [reply packages/mantine-react-table/src/hooks/useMRT_TableInstance.ts]: Yeah, I knew I was out over my skis here, which is where I stopped and solved it another way in my app.

Comment (KevinVandy) [reply packages/mantine-react-table/src/hooks/useMRT_TableInstance.ts]: Just got stuff working. Check out the new dynamic editing and row number example @naughton

**Review by KevinVandy — APPROVED** [review]

**Pull request merged**